src/inputs.rs

```diff
@@ -76,8 +76,13 @@ impl<B: Backend> CompiledCircuit<B> {
             }
 
             (TyKind::Array(el_typ, size), Value::Array(values)) => {
-                if values.len() != (*size as usize) {
-                    panic!("wrong size of array");
+                match size {
+                    ArraySize::Number(n) => {
+                        if values.len() != *n as usize {
+                            panic!("wrong size of array");
+                        }
+                    }
+                    ArraySize::ConstVar(_) => panic!("array argument with size in const var is not supported for main function"),
```

Comment:
in general we should return nicer error to the user if we can :D, here we could even return an error with a span